### PR TITLE
Infighting underwater

### PR DESCRIPTION
**Pull request by Coolthulhu**
Zeds were killing fish, depopulating entire rivers. This should prevent it.

Monsters without MF_SWIMS or MF_AQUATIC have vision range restricted to 1 when under water.
In order to avoid expensive checks for map tile swimmability in sight_range (function used a lot), I moved the `underwater` variable from `player` to `Creature`. It is set when monster voluntarily moves into or is flung into water.

Fixed a bug that allowed monsters to target monsters they can't see.

Re-added fear triggers for nearby player. Now they check for nearby hostiles, like anger triggers.

It may have some simple conflicts with my monster faction PR, but they should be trivially solvable manually.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
Very nice!

File: src/monster.cpp [hunk 1]
```diff
@@ -352,12 +353,13 @@ bool monster::can_drown() const
...
 {
-    return has_flag(MF_DIGS) || (has_flag(MF_CAN_DIG) && g->m.has_flag("DIGGABLE", posx(), posy()));
+    return has_flag(MF_DIGS) || ( has_flag(MF_CAN_DIG) && underwater );
 }
 
 int monster::sight_range( const int light_level ) const
 {
-    if( !can_see() ) {
+    // Non-aquatic monsters can't see much when submerged
+    if( !can_see() || ( underwater && !has_flag( MF_SWIMS ) && !has_flag( MF_AQUATIC ) && !digging() ) ) {
```
Comment (KA101) [inline src/monster.cpp]: Doesn't have to happen *now* but nonaquatic critters also shouldn't be able to smell anything underwater.  (Aquatic critters can smell underwater just fine: see "shark".)

**Comment by KA101** [comment]
Issue: This appears to kill PC sight underwater as well (I'm wearing swimming goggles, still can't see a fungaloid until and unless I bump into it, at which point I attack).  Not so good.

**Comment by KA101** [comment]
OK, let's keep this around.  In the faction-testing build, and milspec turrets didn't even notice me (broad daylight, spring day 1) until I was adjacent.  Nether critter was able to close and attack, though that could be smell.

**Pull request merged**